endpoints/directory_list_generator.py

```diff
@@ -46,6 +47,10 @@ def hello(self, request):
   implemented by HelloService.
   """
 
+  def __init__(self, request=None):
```

Comment:
Are we recreating the generator each time? I suppose it doesn't make much of a difference, but it seems to make more sense to add it as a generate parameter to avoid reconstruction. But not a big deal.